pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1619,32 +1631,32 @@
     ::  +sign-to-json: render sign from request-id as json channel event
     ::
     ++  sign-to-json
+      ~%  %sign-to-json  ..part  ~
       |=  [request-id=@ud =sign:agent:gall]
-      ^-  (unit json)
+      ^-  (unit (quip move json))
       ::  for facts, we try to convert the result to json
       ::
-      =/  jsyn=(unit sign:agent:gall)
-        ?.  ?=(%fact -.sign)       `sign
-        ?:  ?=(%json p.cage.sign)  `sign
+      =/  [from=(unit mark) jsyn=(unit sign:agent:gall)]
+        ?.  ?=(%fact -.sign)       [~ `sign]
+        ?:  ?=(%json p.cage.sign)  [~ `sign]
         ::  find and use tube from fact mark to json
         ::
         =*  have=mark  p.cage.sign
         =*  desc=tape  "from {(trip have)} to json"
-        =/  tube=(unit tube:clay)
-          =/  tuc=(unit (unit cage))
-            (rof ~ %cc [our %home da+now] /[have]/json)
-          ?.  ?=([~ ~ *] tuc)  ~
-          `!<(tube:clay q.u.u.tuc)
-        ?~  tube
-          ((slog leaf+"eyre: no tube {desc}" ~) ~)
-        ::
-        =/  res  (mule |.((u.tube q.cage.sign)))
-        ?:  ?=(%& -.res)
-          `[%fact %json p.res]
-        ((slog leaf+"eyre: failed tube {desc}" ~) ~)
-      ::
+        =/  convert=(unit vase)
+          =/  cag=(unit (unit cage))
+            (rof ~ %cf [our %home da+now] /[have]/json)
+          ?.  ?=([~ ~ *] cag)  ~
+          `q.u.u.cag
+        ?~  convert
+          ((slog leaf+"eyre: no convert {desc}" ~) [~ ~])
+        [`have `[%fact %json (slym u.convert q.q.cage.sign)]]
```

Comment:
This line should have an appropriate `~|` or `~_`, so that it's obvious if it crashes.

Reply:
done